PiwikTracker/PiwikTracker.swift

```diff
@@ -22,6 +22,8 @@ final public class PiwikTracker: NSObject {
     private var queue: Queue
     internal let siteId: String
     
+    public var logger: Logger? = DefaultLogger(minLevel: .warning)
```

Comment:
I would probably make this non optional and not expose the Logger implementations outside the framework.

If the consumer of the framework needs to change the logging level I would hide that behind a method on the PiwikTracker class, e.g. enableDebugLogging that creates and assign new immutable DefaultLogger instance with an appropriate log level.